Generalize Corfunc Slider

jellybean2004 opened the pull request:
## Description

This refactors the Corfunc slider into a reusable component so it can be shared by Invariant, making the slider generic and reusable for components that need different constraint semantics (e.g., Invariant needs to permit extrapolation beyond the raw data).

- Takes `CorfuncSlider.py` out of `src/sas/qtgui/Perspectives/Corfunc`
- Renames to `ExtrapolationSlider.py` and places in `src/sas/qtgui/Utilities`
- Moves utility classes from `src/sas/sascalc/corfunc/calculation_data.py` to `src/sas/sascalc/util.py`
- Updates all imports/usages to match the new locations and names

**Behavioural change (boundaries)**

Introduces two distinct boundary pairs used by different callers:

- Data range — the actual start/end of data points (existing boundary used by Corfunc).
- Extrapolation range — a broader fixed range used by Invariant (10e-5 … 10).

**Usage differences:**

Corfunc: uses Data range only (no functional change except location/imports).
Invariant: uses both:

- Low-Q end and High-Q start are constrained within the Data range.
- High-Q end extends into the Extrapolation range.

Fixes #3824 

## How Has This Been Tested?

Ran the program and tested the slider functionality in both perspectives.

## Review Checklist:

[if using the editor, use `[x]` in place of `[ ]` to check a box]

**Documentation** (check at least one)
- [x] There is **nothing** that needs documenting
- [ ] Documentation changes are **in this PR**
- [ ] There is an **issue** open for the documentation (link?)

**Installers**
- [ ] There is a chance this will affect the **installers**, if so
  - [ ] **Windows** installer (GH artifact) has been tested (installed and worked) 
  - [ ] **MacOSX** installer (GH artifact) has been tested (installed and worked)
  - [ ] **Wheels** installer (GH artifact) has been tested (installed and worked)

**Licensing** (untick if necessary)
- [x] The introduced changes comply with SasView license (BSD 3-Clause)

DrPaulSharp commented on src/sas/qtgui/Perspectives/Corfunc/CorfuncPerspective.py:
```diff
@@ -10,27 +10,26 @@
 from sas.qtgui.Plotting.PlotterData import Data1D
+
+# sas-global
+# pylint: disable=import-error, no-name-in-module
```
We're moving away from using pylint, so could you please remove these comments seeing as you're modifying the line.

jellybean2004 commented:
@DrPaulSharp thank you for your review.

I have updated the slider input validation logic in both Invariant and Corfunc to account for edge cases and disable the buttons if the values are invalid. I have also rewritten the logic in Corfunc to match that of Invariant. So, instead of popping a dialog box on press, the button is just greyed out. What do you think of this?

I will also look into Pydantic a bit more.

DrPaulSharp reviewed (commented):
Ok, I've found another problem a bit like the last one. In the invariant perspective, if you put an invalid value into low-Q end or high-Q start, it snaps back to the min/max allowed value, but the calculate button is not re-enabled.
